doc/api/module.md

```diff
@@ -1048,6 +1048,47 @@ returned object contains the following keys:
 * columnNumber: {number} The 1-indexed columnNumber of the
   corresponding call site in the original source
 
+## TypeScript support
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.0 - Early development
+
+It is possible to execute TypeScript files by by setting the experimental
+flag [`--experimental-strip-types`][].
+Node.js will transpile TypeScript source code into JavaScript source code as `esnext` [ES Modules][].
+During the transpilation process, no type checking is performed, and types are discarded.
+Currently only files with the extension `.ts` are supported,
+meaning it will not support `.mts`, `.cts`, or `.tsx` syntax.
+
+### Limitations
+
+#### TypeScript only features
+
+By design Node.js will not perform transformations on TypeScript only features.
+This means `Enum`, `experimentalDecorators`, `namespaces` and other TS
+only features are not supported.
+You can set `isolatedModules` in your `tsconfig.json` to ensure
+that you are not using TypeScript only features.
+
+Importing [modules without extension][] is not supported.
+
+```ts
+  import { foo } from 'foo'; // it will not work
```

Comment:
I think this should be fixed, at least when importing a ts file from another ts file. I suspect extensions are currently rare in Node projects and I'd like this feature to be able to replace tooling for people eventually. (i.e. shouldUseTSLoader's check should be relaxed)